hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -980,12 +981,18 @@ public void createDirectory(final Path path, final FileSystem.Statistics statist
       if (getAbfsConfiguration().getPrefixMode() == PrefixMode.BLOB) {
         ArrayList<Path> keysToCreateAsFolder = new ArrayList<>();
         checkParentChainForFile(path, tracingContext, keysToCreateAsFolder);
+        boolean blobOverwrite = abfsConfiguration.isEnabledBlobMkdirOverwrite();
 
         HashMap<String, String> metadata = new HashMap<>();
         metadata.put(X_MS_META_HDI_ISFOLDER, TRUE);
+        createFile(path, statistics, blobOverwrite,
+                permission, umask, tracingContext, metadata);
+
         for (Path pathToCreate: keysToCreateAsFolder) {
-          createFile(pathToCreate, statistics, false,
-                  permission, umask, tracingContext, metadata);
+          try {
+            createFile(pathToCreate, statistics, blobOverwrite,
+                    permission, umask, tracingContext, metadata);
+          } catch (FileAlreadyExistsException ex) {}
```

Comment:
should we catch now?
I feel FileAlreadyExist would only come if there is a file on the path. In that case we should propagate back the exception to dev.

Reply:
In that case try catch is not needed is what I feel, the exception would flow back to the caller right ?

Reply:
correct.